synapse/state/__init__.py

```diff
@@ -306,16 +309,11 @@ async def compute_event_context(
             state_group_before_event_prev_group = None
             deltas_to_state_group_before_event = None
 
-            # .. though we need to get a state group for it.
-            state_group_before_event = (
-                await self._state_storage_controller.store_state_group(
-                    event.event_id,
-                    event.room_id,
-                    prev_group=None,
-                    delta_ids=None,
-                    current_state_ids=state_ids_before_event,
-                )
-            )
+            # .. though we need to get a state group for it if we don't have it
+            if not current_state_group:
+                state_group_before_event = None
+            else:
+                state_group_before_event = current_state_group
```

Comment:
Can't we just rename the argument `state_group_to_event`?

Reply:
I think you were getting at something like what I've changed it to: `state_group_before_event = current_state_group`?